R/auditCommonFunctionsBayesian.R

```diff
@@ -19,1941 +19,906 @@
+						unseenLowerBound = lowerBound, unseenUpperBound = upperBound)
+		return(results)
+	}
+
+	results <- list(lowerBound = lowerBound, upperBound = upperBound)
+	return(results)
+}
 
-  shift <- round(posteriorEvidenceRatio / priorEvidenceRatio, 2)
+.jfa.bayesRegression.calculation <- function(options, sample, prevOptions){
+
+	# Bayesian Linear Regression Using the BAS package
+	# This will be replaced by the WASEM technique.
+	sample <- stats::na.omit(sample)
+	
+	sample <- sample[, c(.v(options[["monetaryVariable"]]), .v(options[["auditResult"]]))]
+	colnames(sample) <- c("ist", "soll")
+	formula <- soll ~ ist
+	
+	if(all(sample[,1] == sample[,2])){
+		betas <- c(1, 1, 1)
+	} else {	
+		basResult <- BAS::bas.lm(formula, data = sample)
+		if(options[["areaUnderPosterior"]] == "displayCredibleBound"){
+			basSummary <- BAS:::confint.coef.bas(coef(basResult), level = prevOptions[["confidence"]] - (1 - prevOptions[["confidence"]]))
+		} else {
+			basSummary <- BAS:::confint.coef.bas(coef(basResult), level = prevOptions[["confidence"]])
+		}
+		betas <- as.numeric(basSummary[2, c(1,2, 3)]) 
+	}
+	
+	beta 	<- betas[3]
+	betaMin <- betas[1]
+	betaMax <- betas[2]	
+	n 		<- nrow(sample)
+	N 		<- prevOptions[["populationSize"]]
+	taints 	<- (sample[["ist"]] - sample[["soll"]])/ sample[["ist"]]
+	k 		<- length(which(taints != 0))
+	t 		<- sum(taints)
+	meanB 	<- mean(sample[["ist"]])
+	meanW 	<- mean(sample[["soll"]])
+	corBW 	<- cor(sample[["ist"]], sample[["soll"]])
+	sdW 	<- sd(sample[["auditValue"]])
+
+	pointEstimate <- N * meanW + beta * (prevOptions[["populationValue"]] - N * meanB)
+	stDev <- sdW * sqrt(1 - corBW^2) * ( N / sqrt(n)) * sqrt( (N-n) / (N-1) )
+	
+	lowerBound <- pointEstimate - betaMax * stDev
+	upperBound <- pointEstimate - betaMin * stDev
+
+	results 					<- list()
+	results[["n"]] 				<- as.numeric(n)
+	results[["k"]] 				<- as.numeric(k)
+	results[["t"]] 				<- as.numeric(t)
+	results[["confidence"]] 	<- as.numeric(prevOptions[["confidence"]])
+	results[["method"]] 		<- "regression"
+	results[["popBookvalue"]] 	<- as.numeric(prevOptions[["populationValue"]])
+	results[["pointEstimate"]] 	<- as.numeric(pointEstimate)
+	results[["lowerBound"]] 	<- as.numeric(lowerBound)
+	results[["upperBound"]] 	<- as.numeric(upperBound)
+	results[["materiality"]] 	<- as.numeric(prevOptions[["materiality"]])
+
+	return(results)
+}
 
-  if(shift == "NaN") # Happens when the prior is improper (e.g., Gamma(1, 0))
-    shift <- Inf
+.jfa.bayesCorrelation.calculation <- function(r, n){
 
-  result <- list(shift = shift, 
-                 priorEvidenceRatio = priorEvidenceRatio, 
-                 priorH1 = priorH1, 
-                 priorH0 = priorH0,
-                 postH1 = postH1,
-                 postH0 = postH0,
-                 posteriorEvidenceRatio = posteriorEvidenceRatio)
+	# For the calculation of the Bayes factor for a correlation, we use the standard Bayesian test as described in:
```

Comment:
Why not use the one from Ly, Marsman & Wagenmakers (https://onlinelibrary.wiley.com/doi/full/10.1111/stan.12111)? It's implemented in package `bstats`

Reply:
It would make it consistent with the tests we have in `jaspRegression` and `jaspSummaryStatistics`